src/overlays/actors/ovl_En_Yb/z_en_yb.c

```diff
@@ -37,59 +46,447 @@ const ActorInit En_Yb_InitVars = {
+            // I have taught you, go use it
+            func_801518B0(globalCtx, 0x147C, &this->actor);
+        } else {
+            // regular talk to him first dialogue
+            func_801518B0(globalCtx, 0x147B, &this->actor);
+        }
+    } else if (EnYb_CanTalk(this, globalCtx)) {
+        func_800B8614(&this->actor, globalCtx, 120.0f);
+    }
+
+    if (this->unkFlag410 & 1) {
+        if (!(player->stateFlags2 & 0x8000000)) {
+            this->unkFlag410 &= ~1;
+        }
+    } else if ((player->stateFlags2 & 0x8000000) && this->actor.xzDistToPlayer < 180.0f &&
+               fabsf(this->actor.playerHeightRel) < 50.0f) {
+
+        this->unkFlag410 |= 1;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_SY_TRE_BOX_APPEAR);
+    }
+
+    EnYb_EnableProximityMusic(this);
+}
+
+void EnYb_WaitForMidnight(EnYb* this, GlobalContext* globalCtx) {
+    if (gSaveContext.time < CLOCK_TIME(6, 0)) {
+        EnYb_UpdateAnimation(this, globalCtx);
+        this->alpha += 5;
+        if (this->alpha > 250) {
+            this->alpha = 255;
+            this->actor.flags |= ACTOR_FLAG_1;
+            this->actionFunc = EnYb_Idle;
+        }
+        EnYb_EnableProximityMusic(this);
+    }
+}
+
+void EnYb_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnYb* this = THIS;
+
+    if (CHECK_FLAG_ALL(this->actor.flags, ACTOR_FLAG_1)) {
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+    if (CHECK_FLAG_ALL(this->actor.flags, ACTOR_FLAG_1)) {
+        Actor_MoveWithGravity(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 40.0f, 25.0f, 40.0f, 5);
+    }
+
+    this->actionFunc(this, globalCtx);
+
+    if (this->cutsceneIndex != -1 && ActorCutscene_GetCurrentIndex() != this->cutscenes[this->cutsceneIndex]) {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+            ActorCutscene_SetIntentToPlay(this->cutscenes[this->cutsceneIndex]);
+        } else if (ActorCutscene_GetCanPlayNext(this->cutscenes[this->cutsceneIndex])) {
+            if (this->cutsceneIndex == 0) {
+                ActorCutscene_StartAndSetUnkLinkFields(this->cutscenes[this->cutsceneIndex], &this->actor);
+            }
+        } else {
+            ActorCutscene_SetIntentToPlay(this->cutscenes[this->cutsceneIndex]);
+        }
+    }
+}
+
+void EnYb_PostLimbDrawOpa(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor) {
+    EnYb* this = (EnYb*)actor;
+
+    if (limbIndex == 11) {
+        Matrix_MultiplyVector3fByState(&D_80BFB2F4, &this->actor.focus.pos);
+    }
+    if (limbIndex == 3) {
+        Matrix_MultiplyVector3fByState(&gZeroVec3f, &this->shadowPos);
+    }
+}
+
+void EnYb_PostLimbDrawXlu(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnYb* this = (EnYb*)actor;
```

Comment:
```suggestion
void EnYb_PostLimbDrawXlu(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
    EnYb* this = THIS;
```

Reply:
Oh good grief, _that's_ why the skeleton functions all got actor->thisx renames?

Starting to think nobody really wants to get rid of THIS at all.

Reply:
I'm pretty sure `THIS` will eventually get removed. Most of the skeleton function renames were made by me. I feel the change will be easier if everything is consistently using THIS.

Reply:
I also feel it makes more sense if these limbDraws `actor` variables are named `thisx` so it mimics the four main actor functions, though am happy for discussion if people disagree.

Reply:
No you're right, these skeleton limb/shadow functions always take only one actor as its param (the relevant actor), it has to be type actor because of the function pointer nature but we know what it is, so naming it thisx makes sense.